python-avd/pyavd/_eos_designs/structured_config/underlay/prefix_lists.py

```diff
@@ -21,66 +22,55 @@ class PrefixListsMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> list | None:
-        """Return structured config for prefix_lists."""
-        if self.shared_utils.underlay_bgp is not True and not self.shared_utils.is_wan_router:
-            return None
-
-        if self.shared_utils.overlay_routing_protocol == "none" and not self.shared_utils.is_wan_router:
-            return None
+    @structured_config_contributor
+    def prefix_lists(self: AvdStructuredConfigUnderlayProtocol) -> None:
+        """Set the structured config for prefix_lists."""
+        if not self.shared_utils.is_wan_router and (not self.shared_utils.underlay_bgp or self.shared_utils.overlay_routing_protocol == "none"):
+            return
 
         if not self.inputs.underlay_filter_redistribute_connected:
```

Comment:
let's move this check first as this is just checking inputs and would return immediately without having to check for the more complex ones above

Reply:
Fixed!